pkg/master/controller.go

```diff
@@ -117,6 +117,9 @@ func (c *Controller) UpdateKubernetesService() error {
 		if err := c.CreateMasterServiceIfNeeded("kubernetes", c.ServiceIP, servicePorts, serviceType); err != nil {
 			return err
 		}
+		if err := c.SetService("kubernetes", servicePorts, serviceType); err != nil {
```

Comment:
Maybe a different name and move the logic into `CreateOrUpdate...` ? Something like, if the service already exists make sure it checks out and update it if necessary.